pilot/pkg/security/authz/builder/builder.go

```diff
@@ -352,6 +354,19 @@ func (b Builder) buildTCP(rules *rbacpb.RBAC, shadowRules *rbacpb.RBAC, provider
 	}
 }
 
+func getBadCustomDenyRules(rules *rbacpb.RBAC) *rbacpb.RBAC {
+	rbac := &rbacpb.RBAC{
+		Action:   rbacpb.RBAC_DENY,
+		Policies: map[string]*rbacpb.Policy{},
+	}
+	keys := maps.Keys(rules.Policies)
+	sort.Strings(keys)
+	for i, key := range keys {
+		rbac.Policies[fmt.Sprintf(badCustomActionTemplate, i)] = rules.Policies[key]
```

Comment:
Each rule has a name. Can we append the name with `-deny-due-to-bad-CUSTOM-action`

Reply:
Maybe simply set rules.Action to deny without changing the policy name

Reply:
I think adding a name like `deny-due-to-bad-CUSTOM-action` may be better for users to understand the denied rules, so I prefer the first option.

Reply:
Sounds good

Reply:
The name works. I haven't found any length limit of the policy name, however I'm not sure if the long name like this matters or not:
```json
{
  "name": "envoy.filters.http.rbac",
  "typed_config": {
    "@type": "type.googleapis.com/envoy.extensions.filters.http.rbac.v3.RBAC",
    "rules": {
      "action": "DENY",
      "policies": {
        "istio-ext-authz-ns[istio-system]-policy[ext-authz]-rule[0]-deny-due-to-bad-CUSTOM-action": {
          "permissions": [
            {
              "and_rules": {
                "rules": [
                  {
                    "or_rules": {
                      "rules": [
                        {
                          "url_path": {
                            "path": {
                              "exact": "/headers"
                            }
                          }
                        }
                      ]
                    }
                  }
                ]
              }
            }
          ],
          "principals": [
            {
              "and_ids": {
                "ids": [
                  {
                    "any": true
                  }
                ]
              }
            }
          ]
        }
      }
    }
  }
}

```

Reply:
The naming is already long, not matter.

With this we can know which authz policy not work